realm-jni/src/util.cpp

```diff
@@ -131,6 +131,11 @@ void ThrowException(JNIEnv* env, ExceptionKind exception, const std::string& cla
             jExceptionClass = env->FindClass("java/lang/IllegalStateException");
             message = "Illegal State: " + classStr;
             break;
+
+        case UnreachableVersion:
+            jExceptionClass = env->FindClass("io/realm/internal/async/UnreachableVersionException");
+            message = "Handover failed due to version mismatch: " + classStr;
```

Comment:
This exception is caught inside `RealmQuery` & used by the defined retry policy to decide whether to stop or continue the request.
It can also be propagated to the user if the retry fails. in this case I agree we should come up with something more friendly like "Query failed" or "Query failed due to concurrent modification of the Realm" ?